source/base/DateTime.ooc

```diff
@@ -14,22 +14,34 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-TimeSpan: cover {
-	_ticks: Int64
-	ticks ::= this _ticks
-	init: func@ (=_ticks)
-	kean_base_timeSpan_getTicks: unmangled func -> Int64 { this _ticks }
-	new: unmangled(kean_base_timeSpan_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
-}
+
+import TimeSpan
 
 DateTime: cover {
 	/* Number of 100 ns intervals since 00.00 1/1/1 */
 	_ticks: UInt64
 	ticks ::= this _ticks
-
 	init: func@ (=_ticks)
+	init: func ~fromYearMonthDay (year : Int, month: Int, day: Int) {}
+	init: func ~fromHourMinuteSec (hour : Int, minute : Int, second : Int){
+		if ( isHourValid(hour,minute,second) ){
+			this _ticks = TimeSpan timeToTicks(hour,minute,second)
+		} else {
+			raise ("invalid input specified for constructor(hour,minute,second)")
+		}
+	}
 	isLeapYear: static func (year: Int) -> Bool { (year % 100 == 0) ? (year % 400 == 0) : (year % 4 == 0) }
 	kean_base_dateTime_getTicks: unmangled func -> UInt64 { this _ticks }
 	new: unmangled(kean_base_dateTime_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
 
+	TicksPerMillisecond : static const Int = 1000
+	TicksPerSecond : static const Int = TicksPerMillisecond*1000
+	/* number of days in year ( non-leap ) */
+	DaysInYear : static const Int = 365
+
+	/* validate argument ranges for hour/minutes/seconds vaules */
+	isHourValid : static func (hour : Int, minute : Int, second : Int) -> Bool {
```

Comment:
`isHourValid` sounds like a question... `hourIsValid` or `isValidHour` would be better.